### PR TITLE
Nxp drone boards

### PR DESCRIPTION
**Pull request by han1raaijmakers**
S32K1xx board support has been updated with some bug fixes
Addition of rddrone-uavcan boards based on the S32K1xx processor

### CODE REVIEW

**Comment by patacongo** [comment]
Waiting for David's review.

**Comment by davids5** [comment]
I will have a look at this over the weekend.

File: arch/arm/src/s32k1xx/hardware/s32k1xx_rtc.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#define S32K1XX_RTC_TCR                         (S32K1XX_RTC_BASE + S32K1XX_RTC_TCR_OFFSET)
+#define S32K1XX_RTC_CR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_CR_OFFSET)
+#define S32K1XX_RTC_SR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_SR_OFFSET)
+#define S32K1XX_RTC_LR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_LR_OFFSET)
+#define S32K1XX_RTC_IER                         (S32K1XX_RTC_BASE + S32K1XX_RTC_IER_OFFSET)
+
+/* RTC Register Bitfield Definitions ****************************************************************/
+
+/* TSR Bit Fields */
+#define RTC_TSR_TSR_MASK                         0xFFFFFFFFu
```
Comment (davids5) [inline arch/arm/src/s32k1xx/hardware/s32k1xx_rtc.h]: H
```suggestion
#define RTC_TSR_TSR_MASK                         0xffffffffu
```

**Comment by jarivanewijk** [comment]
I have started to fix some of the code style errors, but I still have many files to check. 

I noticed that a lot of errors have been there since the initial port to S32K. I am not sure how strictly enforced these rules are, and I am trying to fix as many as I can, but for now I have left a lot of the "long line" errors untouched.

**Comment by davids5** [comment]
@jarivanewijk - the nxstyle tool was updated and is finding more things now. -m 86 is the default for line length.  

The nxstyle output is now formatted like compiler output so the changes can be quickly done in eclipse or any editor that parses compiler output.  (there is a close PR with the setup details in this repo)

The idea of the comment blocks is as @patacongo calls it "a ruler". Consistency on comment block length is the goal.  In a arch header file longer lines are acceptable if consistent.  In c files wrapping at 86 is the goal. 

I am hopeful that some day NuttX will have tool like Astyle, Uncrustify or Clang format, but for now it is still a tad to ambiguous for comfort.

**Comment by patacongo** [comment]
@davids5 "he nxstyle tool was updated and is finding more things now. -m 86"  This is no longer true.

The tool now automatically calculates the line width from the block comments.  For most files, that will be 78.  -m 86 should NOT be used.  That used to be the maximum line width but now has a different meaning.  It is NOW the line width increment.  If you wanted the usual -m86 in the past, you would now need to select -m 8 because 78+8 = 86

**Comment by jarivanewijk** [comment]
So... We should stick to line length 78? (Perhaps with an occasional extra character or two if there's no nice way to split the line into two?)

**Comment by patacongo** [comment]
@jarivanewijk "So... We should stick to line length 78?"  Remeber, -m does not select the line width.  That is no longer under user control.  nxstyle will parts all of the block comments and determine the correct line with.    The -m value allows a little "slop" at the end of the line, but is not the line width.  -m should be set to -m 8 for backward compatibility (i.e., allow lines to extend up to 8 characters beyond the block comments.)

"(Perhaps with an occasional extra character or two if there's no nice way to split the line?)" -- Yes that is why you need to allow a little slop.  So lines just do not divide nicely.

**Comment by davids5** [comment]
@patacongo - what is the requirement to pass coding standard check or said another way what is now considered acceptable? 

Is it up to 8 characters past the ruler (comment block length)?  Why is it not the longest line of code (or define) that sets the ruler length?

**Comment by patacongo** [comment]
The "ruler length" must be the same on every block comment, otherwise an error will be generated.  Lines of code must be less than or equal to the ruler length (plus -m slop).  Given the idea of "the longest line of code" makes no sense to me.

**Comment by jarivanewijk** [comment]
I have gone through all the files that were already part of this PR and fixed most of the code style errors. However, for some reason I am now unable to build any configuration, even configurations that we never touched.

$ tools/configure.sh s32k144evb:nsh
  Copy files
  Select CONFIG_HOST_LINUX=y
  Refreshing...
make: *** [do_olddefconfig] Error 1
ERROR: failed to refresh

**Comment by jarivanewijk** [comment]
Nevermind, we forgot to remove a few boards from the boards/Kconfig file. Their board configuration is not included in this PR. I think everything is okay now.

**Comment by patacongo** [comment]
@davids5 Please merge this when it is ready.

**Comment by jarivanewijk** [comment]
@davids5 - As discussed, the branch was again rebased onto latest master. The defconfigs were still up-to-date and no additional style fixes were needed since last week.

Please review this so we can get this monster of a PR merged.

In particular, have a look at the few files in /arch/arm/src that do not belong to the s32k1xx arch. We fixed a few small bugs for kinetis and imxrt that we also found in s32k1xx arch.
Additionally, please have a quick look at the board configurations that we added. We would like to know if in general everything is in the right place.

File: arch/arm/src/s32k1xx/hardware/s32k1xx_rtc.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+#define S32K1XX_RTC_CR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_CR_OFFSET)
+#define S32K1XX_RTC_SR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_SR_OFFSET)
+#define S32K1XX_RTC_LR                          (S32K1XX_RTC_BASE + S32K1XX_RTC_LR_OFFSET)
+#define S32K1XX_RTC_IER                         (S32K1XX_RTC_BASE + S32K1XX_RTC_IER_OFFSET)
+
+/* RTC Register Bitfield Definitions ****************************************************************/
+
+/* TSR Bit Fields */
+
+#define RTC_TSR_TSR_MASK                         0xffffffffu
```
Comment (davids5) [inline arch/arm/src/s32k1xx/hardware/s32k1xx_rtc.h]: @jarivanewijk This is not consistent with the way the defines are in the archs/

Please re-formatted this file to consistent  with the [others](https://github.com/apache/incubator-nuttx/blob/master/arch/arm/src/s32k1xx/hardware/s32k1xx_flexcan.h#L334-L349)

File: arch/arm/src/s32k1xx/s32k1xx_enet.c [hunk 3]
```diff
@@ -222,7 +231,7 @@
  *             = 23
  */
 
-#define S32K1XX_MII_SPEED  0x38 /* 100Mbs. Revisit and remove hardcoded value */
+#define S32K1XX_MII_SPEED  0x0F /* 100Mbs. Revisit and remove hardcoded value */
```
Comment (davids5) [inline arch/arm/src/s32k1xx/s32k1xx_enet.c]:
```suggestion
#define S32K1XX_MII_SPEED  0x0f /* 100Mbs. Revisit and remove hardcoded value */
```

File: boards/arm/s32k1xx/s32k144evb/include/board.h [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+#define PIN_LPSPI2_SCK   PIN_LPSPI2_SCK_2   /* PTE15 */
+#define PIN_LPSPI2_MISO  PIN_LPSPI2_SIN_2   /* PTE16 */
+#define PIN_LPSPI2_MOSI  PIN_LPSPI2_SOUT_1  /* PTA8  */
+#define PIN_LPSPI2_PCS   PIN_LPSPI2_PCS0_2	/* PTA9  */
+
+/* I2C selections ***********************************************************/
+
+#define PIN_LPI2C0_SCL   PIN_LPI2C0_SCL_2   /* PTA3 */				
+#define PIN_LPI2C0_SDA	 PIN_LPI2C0_SDA_2   /* PTA2 */
```
Comment (davids5) [inline boards/arm/s32k1xx/s32k144evb/include/board.h]:
```suggestion
#define PIN_LPI2C0_SDA   PIN_LPI2C0_SDA_2   /* PTA2 */
```

**Comment by davids5** [comment]
This PR was split into 3 PR. Close in Favor if #224 

imxrt added missing i2c prescale mask #222  Merged
Kinetis renamed TJA1100 to TJA110X registers #223  Merged
S32K add support for Nxp drone boards #224 Under Revire

**Pull request closed without merging**